pkg/server/middlewares.go

```diff
@@ -42,18 +43,23 @@ func (s *Server) MiddlewareRequireClientCertificate(c *gin.Context) {
 // This middleware should be used first in the chain.
 func (s *Server) MiddlewareProxyHeaders(c *gin.Context) {
 	// Ensure required headers are present
-	for i := range proxyHeaders {
-		if c.Request.Header.Get(proxyHeaders[i]) == "" {
-			AbortWithError(c, NewResponseErrorf(http.StatusBadRequest, "Missing header %s", proxyHeaders[i]))
+	for _, header := range proxyHeaders {
+		if c.Request.Header.Get(header) == "" {
+			AbortWithError(c, NewResponseErrorf(http.StatusBadRequest, "Missing header %s", header))
 			return
 		}
 	}
 
+	// Get the X-Forwarded-For header
+	xForwardedFor := c.Request.Header.Get("X-Forwarded-For")
+	xForwardedPort := c.Request.Header.Get("X-Forwarded-Port")
+
+	// Split the X-Forwarded-For header to get the originating client IP
+	ips := strings.Split(xForwardedFor, ",")
+	clientIP := strings.TrimSpace(ips[0])
```

Comment:
While this code is correct, I would recommend here a slightly different approach for performance reasons, since this code is in the hot path of every request.

`strings.Split` iterates through the entire string and allocates a slice with each element. In Go, heap allocations are some of the worst offenders when it's about performance, since they also increase the load on the garbage collector.

Since we only care about the first result, I would recommend using [`strings.Cut`](https://pkg.go.dev/strings#Cut) instead. Something like:

```suggestion
	clientIP, _, _ := strings.Cut(xForwardedFor, ",")
	clientIP = strings.TrimSpace(clientIP)
```

Reply:
Wow many thanks! 
I wasn't even aware of `strings.Cut`, I appreciate the detailed explanation and the suggested change!